components/rmrk/Gallery/CollectionItem.vue

```diff
@@ -257,6 +278,31 @@ export default class CollectionItem extends mixins(ChainMixin, PrefixMixin) {
     })
   }
 
+  protected async fetchNFTStats() {
+    try {
+      const query =
+        this.urlPrefix === 'rmrk'
+          ? await import('@/queries/nftStatsByIssuer.graphql')
+          : await import('@/queries/unique/nftStatsByIssuer.graphql')
+      this.$apollo.addSmartQuery('collections', {
+        query: query.default,
+        manual: true,
+        client: this.urlPrefix,
+        loadingKey: 'isLoading',
+        result: this.handleMintTimeResult,
+        variables: () => {
+          return {
+            account: this.issuer || '',
+          }
+        },
+        fetchPolicy: 'cache-and-network',
+      })
+    } catch (e) {
+      showNotification(`${e}`, notificationTypes.danger)
+      console.warn(e)
+    }
+  }
```

Comment:
Isn't this duplicate from another component?

Reply:
Yes, many codes from "IdentityPopover.vue".
What's the problem? Better to refactor this for both components?

Reply:
> 

I can use **_mixin_** to refactor them

Reply:
Why such a huge query?